app/forms/claims/claim/mentor_training_form.rb

```diff
@@ -62,11 +60,11 @@ def custom_hours_selected?
   end
 
   def max_hours
-    @max_hours ||= Claims::Mentor::CalculateRemainingMentorTrainingHoursForProvider.call(mentor:, provider:, mentor_training:)
+    @max_hours ||= training_allowance.remaining_hours
   end
 
   def max_hours_equals_maximum_claimable_hours?
-    max_hours == MAXIMUM_CLAIMABLE_HOURS_PER_PROVIDER
+    max_hours == training_allowance.total_hours
   end
```

Comment:
I appreciate this is probably out of scope for this PR, but I wanted to point it out nonetheless. I'm not a fan of these method names. I think we could improve upon them.

### `#max_hours`

`max_hours` is just `remaining_hours` under a different name. It doesn't seem to be a meaningful or intentional change in terminology – in fact, I think it's unnecessarily confusing.

I'd suggest we just delegate this method to the underlying training allowance object and avoid the name change altogether. Especially because the underlying `remaining_hours` method is already memoized – there's no reason _not to_ delegate it.

```ruby
delegate :remaining_hours, to: :training_allowance
```

### `#max_hours_equals_maximum_claimable_hours?`

This method name seems doubly confusing – _"What's the difference between 'max hours' and 'maximum claimable hours'? They sound like the same thing. And what does it mean if they're equal?"_.

This method is **only used in one place** in the app. It's used to decide whether this component gets rendered:

https://github.com/DFE-Digital/itt-mentor-services/blob/2bc7df6e8a739e45961c27bc46104e78cc634740/app/components/claims/claim/mentor_training_form/disclaimer_component.rb#L17-L19

I'd suggest a more meaningful method would be `some_hours_already_claimed?`, to indicate that some hours have already been claimed – in other words, because the 'remaining hours' is less than 'total hours' of the training allowance. Which would be a 'positive' use of logic in the component (i.e. we wouldn't need to `!` flip the boolean value):

```diff
def render?
-  !mentor_training_form.max_hours_equals_maximum_claimable_hours?
+  mentor_training_form.some_hours_already_claimed?
end
```

Additionally, I'm not _convinced_ this method _really belongs_ in the form object. Since the the component is the only thing that uses the method, is there really a legitimate need for it elsewhere?

If it's needed elsewhere, it should probably live directly on the `Claims::TrainingAllowance` object.

If it's only needed by the component itself, then the logic should probably just live directly inside the `#render?` method.

---

These would be nice improvements to make to the code, but as I said I appreciate they might not be technically in scope for this PR.

Reply:
Follow-up PR created to address this #1086